Blog - How To Build A PowerBI Report For Octopus Deploy

aspnerd opened the pull request.

wordlee commented:
H @aspnerd. That's for submitting. Cool use of data! I've committed some initial edits (I'll take another pass later in the week). I condensed the intro a bit, we try to put what the reader will learn in the first paragraph or so. I've also simplified the language in a few places to keep the tone consistent with the rest of the blog. 
In the ROI section, I removed the Octopus pricing, I see why you added it and I agree it's great value, but it felt a little bit salesy. I hope that makes sense.

aspnerd commented:
> H @aspnerd. That's for submitting. Cool use of data! I've committed some initial edits (I'll take another pass later in the week). I condensed the intro a bit, we try to put what the reader will learn in the first paragraph or so. I've also simplified the language in a few places to keep the tone consistent with the rest of the blog.
> In the ROI section, I removed the Octopus pricing, I see why you added it and I agree it's great value, but I felt it could read as being a little bit salesy. I hope that makes sense.

I think we need to at least keep something in there regarding the cost of a person vs. the license of Octopus Deploy...this is a major hurdle for companies in my experience and they can't justify the cost, yet if they compared the cost to a full or part-time person it would come in perspective.

aspnerd commented:
> H @aspnerd. That's for submitting. Cool use of data! I've committed some initial edits (I'll take another pass later in the week). I condensed the intro a bit, we try to put what the reader will learn in the first paragraph or so. I've also simplified the language in a few places to keep the tone consistent with the rest of the blog.
> In the ROI section, I removed the Octopus pricing, I see why you added it and I agree it's great value, but I felt it could read as being a little bit salesy. I hope that makes sense.

I think we need to at least keep something in there regarding the cost of a person vs. the license of Octopus Deploy...this is a major hurdle for companies in my experience and they can't justify the cost, yet if they compared the cost to a full or part-time person it would come in perspective.

wordlee commented:
> > H @aspnerd. That's for submitting. Cool use of data! I've committed some initial edits (I'll take another pass later in the week). I condensed the intro a bit, we try to put what the reader will learn in the first paragraph or so. I've also simplified the language in a few places to keep the tone consistent with the rest of the blog.
> > In the ROI section, I removed the Octopus pricing, I see why you added it and I agree it's great value, but I felt it could read as being a little bit salesy. I hope that makes sense.
> 
> I think we need to at least keep something in there regarding the cost of a person vs. the license of Octopus Deploy...this is a major hurdle for companies in my experience and they can't justify the cost, yet if they compared the cost to a full or part-time person it would come in perspective.

Hi Jeff, at the bottom of each post, we have a call out that states that Octopus Server and Cloud is free for small teams. I think that combined with the savings you mentioned makes a really strong case.

aspnerd commented:
@wordlee I'm not sure I like the opening sentence now.  It's not about the templates, it's about the data, what it can show you, and time saved through automation.

wordlee commented on blog/2020-06/how-to-build-a-powerbi-report-for-octopus-deploy/index.md:
```diff
@@ -0,0 +1,215 @@
+  - Engineering
+---
+
+In this post, I share some templates that will help you build a PowerBI report from your Octopus Deploy database that will show you how much time and money automation has saved you and your company. I’ll also show you how to highlight development issues, such as how many times releases have been skipped due to bugs not being found in lower level environments.
```
```suggestion
In this post, I share some data I pulled from my Octopus Deploy database for use in a PowerBI report. I use these reports to highlight how much time and money automation has saved my company, and to spot potential development issues that slow down deployments, such as how many times releases have been skipped due to bugs not being found in lower-level environments. At the end of the post, I share some templates to make this process easier for you to replicate.
```

aspnerd commented:
@wordlee better...I'd still change this line... "At the end of the post, I share some templates to make this process easier for you to replicate." to something like this... "At the end of this post, I share the templates I created so you don't have to start from scratch."

aspnerd commented:
Also, are these highlighted items supposed to appear like this? 

![image](https://user-images.githubusercontent.com/17560259/85335191-9413ff80-b4aa-11ea-817d-6742b4b710e5.png)

wordlee commented:
> Also, are these highlighted items supposed to appear like this?
> 
> ![image](https://user-images.githubusercontent.com/17560259/85335191-9413ff80-b4aa-11ea-817d-6742b4b710e5.png)

They do in GitHub preview, but they're transformed on the blog. {#test} becomes an anchor for the heading, and ::: becomes a highlighted callout to draw attention to the text :)

wordlee commented on blog/2020-06/how-to-build-a-powerbi-report-for-octopus-deploy/index.md:
```diff
@@ -0,0 +1,215 @@
+  - Engineering
+---
+
+In this post, I share some data I pulled from my Octopus Deploy database for use in a PowerBI report. I use these reports to highlight how much time and money automation has saved my company, and to spot potential development issues that slow down deployments, such as how many times releases have been skipped due to bugs not being found in lower-level environments. At the end of the post, I share ...
```
```suggestion
In this post, I share some data I pulled from my Octopus Deploy database for use in a PowerBI report. I use these reports to highlight how much time and money automation has saved my company, and to spot potential development issues that slow down deployments, such as how many times releases have been skipped due to bugs not being found in lower-level environments. At the end of this post, I share the templates I created so you don't have to start from scratch.
```

aspnerd commented:
@wordlee I adjusted it a little more....please let me know how this looks/sounds do you.

In this post, I share the analysis of the data I pulled from my Octopus Deploy database using PowerBI Desktop. I was able to highlight how much time and money automation has saved my company, and spot potential development issues that slow down deployments, such as how many times releases have been skipped due to bugs not being found in lower-level environments. At the end of this post, I share the templates I created so you don't have to start from scratch.

Hopefully, these reports will get you pointed in the right direction to help you highlight how much automation can save in time and money, and even justify the need for automation and testing resources.

wordlee commented:
> @wordlee I adjusted it a little more....please let me know how this looks/sounds do you.
> 
> In this post, I share the analysis of the data I pulled from my Octopus Deploy database using PowerBI Desktop. I was able to highlight how much time and money automation has saved my company, and spot potential development issues that slow down deployments, such as how many times releases have been skipped due to bugs not being found in lower-level environments. At the end of this post, I share the templates I created so you don't have to start from scratch.
> 
> Hopefully, these reports will get you pointed in the right direction to help you highlight how much automation can save in time and money, and even justify the need for automation and testing resources.

That's great @aspnerd!

aspnerd commented:
@wordlee when will this go live?

wordlee commented:
@aspnerd adding @robpearson for that question :)

robpearson commented:
@aspnerd I currently have it scheduled to go live on July 15th. I need to request a custom graphic for it. It's a great post! :)

aspnerd commented:
> @aspnerd I currently have it scheduled to go live on July 15th. I still need to request a custom graphic for it. It's a great post! :)

cool...I was thinking about that...I had an idea hit me... Einstein Octopus (Octopus with Einstein's crazy hair).... with calculations coming off a clipboard with a pie chart and numbers? just an idea...
